client/app/routes/userpage.tsx

```
@@ -53,27 +55,23 @@ export default function UserPage({ loaderData }: Route.ComponentProps) {
         <TabsContent value="posts">
           <h2>Posts</h2>
           {postData.length > 0 ? (
-            <div className="posts-list">
+            <div className="posts-list flex flex-col gap-5">
               {postData.map((post: PostData) => (
-                <div key={post.id} className="post-card">
-                  <p>{post.username}</p>
-                  <h3>{post.title}</h3>
-                  <p>{post.text}</p>
-                </div>
+                <Post post={post}/>
```

Biome (1.9.4)

[error] Missing key property for this element in iterable.

The order of the items may change, and having a key can help React identify which item was moved.
Check the React documentation. 

(lint/correctness/useJsxKeyInIterable)